File: flutter/crowdsourcing/lib/models/object/Location.dart
class Location {
  String name, phone, province, city, plot, street, others,town;
  int id;
  bool isMain;
  static const String Id = "id";
  static const String Name = 'name';
  static const String Phone = 'number';
  static const String Province = 'province';
  static const String City = 'city';
  static const String Plot = "plot";
  static const String Town = 'town';
  static const String Street = "street";
  static const String IsMain = "main";
  static const String Others = "others";

  Location(
      {this.id,
      this.name,
      this.phone,
      this.province,
      this.city,
      this.plot,
        this.town,
      this.street,
      this.others,
      this.isMain});

  static Location fromJsonMap(Map<String, dynamic> map) {
    if (map == null) return null;
    return Location(
        id: map[Id],
        town: map[Town],
        name: map[Name],
        phone: map[Phone],
        province: map[Province],
        city: map[City],
        plot: map[Plot],
        others: map[Others],
        isMain: map[IsMain],
        street: map[Street]);
  }

  Map toJson() => {
        Id: id,
        Phone: phone,
        Province: province,
        Name: name,
        City: city,
        Plot: plot,
        Town:town,
        Others: others,
        Street: street,
        IsMain: isMain
      };

  @override
  toString() {
    return this.province == null
        ? "暂无地址设置"
        : this.province + (this.city??"") + (this.plot??"") + (this.town??"")+this.others??"";
  }
}

File: flutter/crowdsourcing/android/app/src/main/kotlin/com/example/crowdsourcing/model/Location.java
package com.example.crowdsourcing.model;

import com.google.gson.Gson;

public class Location {
    private static final long serialVersionUID = 1L;




    private String province;

    private String  city;

    private String  street;

    private String  others;

    private String plot;
    private  String town;



    public Location(){

    }




    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getOthers() {
        return others;
    }

    public void setOthers(String others) {
        this.others = others;
    }


    public String getProvince() {
        return province;
    }

    public void setProvince(String province) {
        this.province = province;
    }

    public String getPlot() {
        return plot;
    }

    public void setPlot(String plot) {
        this.plot = plot;
    }


    public  String toString(){
        return  new Gson().toJson(this);
    }


    public String getTown() {
        return town;
    }

    public void setTown(String town) {
        this.town = town;
    }
}
